modules/convert/convert.go

```diff
@@ -30,28 +30,76 @@ func ToEmail(email *models.EmailAddress) *api.Email {
 }
 
 // ToBranch convert a git.Commit and git.Branch to an api.Branch
-func ToBranch(repo *models.Repository, b *git.Branch, c *git.Commit, bp *models.ProtectedBranch, user *models.User) *api.Branch {
+func ToBranch(repo *models.Repository, b *git.Branch, c *git.Commit, bp *models.ProtectedBranch, user *models.User, isRepoAdmin bool) *api.Branch {
 	if bp == nil {
 		return &api.Branch{
-			Name:                b.Name,
-			Commit:              ToCommit(repo, c),
-			Protected:           false,
-			RequiredApprovals:   0,
-			EnableStatusCheck:   false,
-			StatusCheckContexts: []string{},
-			UserCanPush:         true,
-			UserCanMerge:        true,
+			Name:                          b.Name,
+			Commit:                        ToCommit(repo, c),
+			Protected:                     false,
+			RequiredApprovals:             0,
+			EnableStatusCheck:             false,
+			StatusCheckContexts:           []string{},
+			UserCanPush:                   true,
+			UserCanMerge:                  true,
+			EffectiveBranchProtectionName: "",
+			EffectiveBRanchProtectionID:   0,
 		}
 	}
+	branchProtectionName := ""
+	var branchProtectionID int64
+	if isRepoAdmin {
+		branchProtectionName = bp.BranchName
```

Comment:
There's surely something I'm missing here. In what case is `bp.BranchName` different from `b.Name`?

Reply:
If we support wildcard protection it could differ. But I could remove it for now.

Reply:
No, it's OK. I wasn't sure I was reading this properly.